AK/Format.h

```diff
@@ -270,14 +269,19 @@ class FormatBuilder {
 
 class TypeErasedFormatParams {
 public:
-    ReadonlySpan<TypeErasedParameter> parameters() const { return m_parameters; }
+    TypeErasedFormatParams(u32 size)
+        : m_size(size)
+    {
+    }
+
+    ReadonlySpan<TypeErasedParameter> parameters() const { return { m_parameters, m_size }; }
 
-    void set_parameters(ReadonlySpan<TypeErasedParameter> parameters) { m_parameters = parameters; }
     size_t take_next_index() { return m_next_index++; }
 
 private:
-    ReadonlySpan<TypeErasedParameter> m_parameters;
-    size_t m_next_index { 0 };
+    u32 m_size { 0 };
+    u32 m_next_index { 0 };
+    TypeErasedParameter m_parameters[0];
```

Comment:
I remember us using unknown length arrays so just `[]`, which is a c extension instead, not sure what is to be preferred

Reply:
That does not compile
```
AK/Format.h:297:37: error: base class 'TypeErasedFormatParams' has a flexible array member
  297 | class VariadicFormatParams : public TypeErasedFormatParams {
      |                                     ^
AK/Format.h:302:11: error: type 'TypeErasedFormatParams' is not a direct or virtual base of 'VariadicFormatParams<allow_debug_formatters, Parameters...>'
  302 |         : TypeErasedFormatParams(sizeof...(Parameters))
      |           ^~~~~~~~~~~~~~~~~~~~~~
```